geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/QueryBuilder.scala

```diff
@@ -0,0 +1,196 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.memory.cqengine.utils
+
+import com.googlecode.cqengine.attribute.Attribute
+import com.googlecode.cqengine.query.simple.{Between, GreaterThan, LessThan, SimpleQuery}
+import org.locationtech.geomesa.filter.PropertyLiteral
+import org.opengis.feature.simple.SimpleFeature
+
+trait CompQueryBuilder {
+  // The type the product queries process, and its class object
+  type VALUE <: Comparable[VALUE]
+  val valueClass: Class[VALUE]    // needed to interact with gt's Java API.
```

Comment:
Yeah, I think that ``ClassTag`` is one of the things that we tried; we ran into issues with the ``Comparable`` trait, but I don't know if we tried this specifically. In a follow-on ticket: https://geomesa.atlassian.net/browse/GEOMESA-1408.